talkvoter/v1_resources.py

```diff
@@ -1,22 +1,70 @@
 from flask_restful import Resource, Api
-from flask import Blueprint
+from flask import Blueprint, abort
+from sqlalchemy.sql.expression import func
 from marshmallow import ValidationError
-from webargs.flaskparser import use_args
-from .models import db
-from .serializers import VoteSchema
+from .models import db, Talk
+from .serializers import VoteSchema, TalkSchema
 from .constants import VoteValue
 
 
 api_bp = Blueprint('api_v1', __name__)
 api = Api(api_bp)
 
 
+class TalksResource(Resource):
+
+    def get(self):
+        schema = TalkSchema()
+        data = {}
+        talk_objs = db.session.query(Talk)
+        ret_code = 200
+        data = schema.dump(talk_objs, many=True).data
+        return data, ret_code
+
+
+api.add_resource(TalksResource, '/talks/', endpoint="api.talks")
+
+
+class TalkResource(Resource):
+
+    def get(self, id):
+        schema = TalkSchema()
+        data = {}
+        talk_obj = db.session.query(Talk).filter(Talk.id == id).first()
```

Comment:
There is a flask plugin for SQL alchemy that has *_or_404() helpers

http://flask-sqlalchemy.pocoo.org/2.3/api/

This works so I'm good either way.

Reply:
Interesting, I functionalized this logic, and then I saw your comment.  Do you know if that _or_404 method can take a custom error message?

Reply:
Not that I recall. Think it just bubbles up a `NotFound` exception which Flask turns into a 404.

I like the flexibility of having more info and honestly, like most flask plugins, it's saving us from writing 10-20 lines of code. This looks good to me!